packages/web3-utils/src/validation.ts

```diff
@@ -134,10 +134,10 @@ export const compareBlockNumbers = (blockA: BlockNumberOrTag, blockB: BlockNumbe
 	) {
 		return 0;
 	}
-	if (blockA === 'earliest' && blockB > 0) {
```

Comment:
I do not clearly understand how it is related to the typescript version.

Reply:
The new TypeScript version detected an error in comparing not a necessarily number with a number.   And in this PR I optimized the condition.